cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -0,0 +1,151 @@
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+import java.nio.channels.FileChannel
+import java.nio.file.{FileSystems, Path, StandardOpenOption}
+import java.nio.channels.FileChannel.MapMode
+import java.io.{FileOutputStream, DataOutputStream, BufferedOutputStream, File}
+
+import com.twitter.cassovary.util.NodeNumberer
+import com.twitter.cassovary.util.io.{AdjacencyListGraphReader, MemoryMappedIntLongSource, IntLongSource}
+import com.twitter.util.Stopwatch
+
+/**
+ * A graph which reads edge data from a memory mapped file.  There is no object overhead per node: the memory
+ * used for n nodes and m edges with both in-neighbor and out-neighbor access is exactly 8 + 16*n + 8*m bytes.
+ * Also, Loading is very fast because no parsing of text is required.  Loading time is exactly the time it takes the operating system
+ * to map data from disk into memory.
+ *
+ * Currently only supports storing both in-neighbors and out-neighbors of nodes (StoredGraphDir.BothInOut).
+ * The binary format is currently subject to change.
+ */
+
+/* Storage format
+byteCount  data
+4          (reserved, later use for versioning or indicating undirected vs directed)
+4          n (i. e. the number of nodes)
+8*(n+1)    Offsets into out-neighbor data. Index i (a Long) points to the out-neighbor data of node i. Index n is needed
```

Comment:
Good point.  I added a comment to the javadoc clarifying nodes are 0 to nodeCount - 1, and changed the comment to "Index n is needed to compute the outdegree of node n - 1."